src/management/studio.ts

```diff
@@ -0,0 +1,158 @@
+    } else {
+      await writeStudioProjectToConfigStore(options, studioWorkspace.firebaseProjectId);
+      return studioWorkspace.firebaseProjectId;
+    }
+  }
+  // Otherwise, Studio and the CLI agree
+  return studioWorkspace.firebaseProjectId;
+}
+
+export interface StudioWorkspace {
+  name: string;
+  firebaseProjectId: string | undefined;
+}
+
+async function getStudioWorkspace(): Promise<StudioWorkspace | undefined> {
+  const workspaceId = process.env.WORKSPACE_SLUG;
+  if (!workspaceId) {
+    logger.error(
+      `Failed to fetch Firebase Project from Studio Workspace because WORKSPACE_SLUG environment variable is empty`,
+    );
+    return undefined;
+  }
+  try {
+    const res = await studioClient.request<void, StudioWorkspace>({
+      method: "GET",
+      path: `/workspaces/${workspaceId}`,
+      timeout: TIMEOUT_MILLIS,
+    });
+    return res.body;
+  } catch (err: any) {
+    let message = err.message;
+    if (err.original) {
+      message += ` (original: ${err.original.message})`;
+    }
+    logger.error(`Failed to fetch Firebase Project from current Studio Workspace: ${message}`);
+    // We're going to fail gracefully so that the caller can handle the error
+    return undefined;
+  }
+}
+
+async function writeStudioProjectToConfigStore(options: Options, studioProjectId: string) {
+  if (options.projectRoot) {
+    logger.info(
+      `Updating Firebase CLI active project to match Studio Workspace '${studioProjectId}'`,
+    );
+    utils.makeActiveProject(options.projectRoot, studioProjectId);
+    recordStudioProjectSyncTime();
+  }
+}
+
+/**
+ * Sets the active project for the current Firebase Studio Workspace
+ * @param projectId The project ID saved in spanner
+ * @return A promise that resolves when complete
+ */
+export async function updateStudioFirebaseProject(projectId: string): Promise<void> {
+  logger.info(`Updating Studio Workspace active project to match Firebase CLI '${projectId}'`);
+  const workspaceId = process.env.WORKSPACE_SLUG;
+  if (!workspaceId) {
+    logger.error(
+      `Failed to update Firebase Project for Studio Workspace because WORKSPACE_SLUG environment variable is empty`,
+    );
+    return;
+  }
+  try {
+    await studioClient.request({
+      method: "PATCH",
+      path: `/workspaces/${workspaceId}`,
+      responseType: "json",
+      body: {
+        firebaseProjectId: projectId,
+      },
+      queryParams: {
+        updateMask: "workspace.firebaseProjectId",
+      },
+      timeout: TIMEOUT_MILLIS,
+    });
+  } catch (err: any) {
+    let message = err.message;
+    if (err.original) {
+      message += ` (original: ${err.original.message})`;
+    }
+    logger.error(
+      `Failed to update active Firebase Project for current Studio Workspace: ${message}`,
+    );
+  }
+  recordStudioProjectSyncTime();
+}
+
+/**
+ * Records the last time we synced the Studio project in Configstore.
+ * Conviently, this triggers the file watcher.
```

Comment:
Suggestion: "This line is important to trigger a file watcher in Firebase Studio that keeps the UI in sync"

I think a future reader will may not have context